cmd/config/apply.go

```diff
@@ -0,0 +1,154 @@
+package config
+
+import (
+	"context"
+	"fmt"
+	"github.com/armory/armory-cli/pkg/cmdUtils"
+	"github.com/armory/armory-cli/pkg/config"
+	"github.com/armory/armory-cli/pkg/configCmd"
+	"github.com/armory/armory-cli/pkg/model"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+	"time"
+)
+
+const (
+	configApplyShort   = "Sync configuration file with Armory CD-as-a-Service"
+	configApplyLong    = "Sync configuration file with Armory CD-as-a-Service"
+	configApplyExample = "armory config apply [options]"
+)
+
+type configApplyOptions struct {
+	configFile string
+}
+
+func NewConfigApplyCmd(configuration *config.Configuration) *cobra.Command {
+	options := &configApplyOptions{}
+	cmd := &cobra.Command{
+		Use:     "apply --file [<path to file>]",
+		Aliases: []string{"apply"},
+		Short:   configApplyShort,
+		Long:    configApplyLong,
+		Example: configApplyExample,
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			cmdUtils.ExecuteParentHooks(cmd, args)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return apply(cmd, options, configuration)
+		},
+	}
+	cmd.Flags().StringVarP(&options.configFile, "file", "f", "", "path to the configuration file")
+	err := cmd.MarkFlagRequired("file")
+	if err != nil {
+		return nil
+	}
+	return cmd
+}
+
+func apply(cmd *cobra.Command, options *configApplyOptions, configuration *config.Configuration) error {
+	payload := model.ConfigurationConfig{}
+	//in case this is running on a GitHub instance
+	gitWorkspace, present := os.LookupEnv("GITHUB_WORKSPACE")
+	_, isATest := os.LookupEnv("ARMORY_CLI_TEST")
```

Comment:
This seems like a smell, but I am not confident enough in Golang yet to say for certain

Reply:
If you have to take this approach I would add a private field to `configApplyOptions` that has a comment saying that it should only be used for tests. I think that is a more common pattern for Go.

Reply:
Fair, we use the above pattern in multiple locations throughout the project so I opened a ticket to refactor. 
https://armory.atlassian.net/browse/CDAAS-1633